src/library/scala/Array.scala

```diff
@@ -61,7 +61,35 @@ object Array extends FallbackArrayBuilding {
   val emptyShortArray   = new Array[Short](0)
   val emptyObjectArray  = new Array[Object](0)
 
-  implicit def canBuildFrom[T](implicit t: ClassTag[T]): CanBuildFrom[Array[_], T, Array[T]] =
+  implicit def canBuildFrom[T](implicit t: ClassTag[T]): CanBuildFrom[Array[_], T, Array[T]] = {
+    val tag = implicitly[ClassTag[T]]
+    (tag.runtimeClass match {
+      case java.lang.Byte.TYPE      => cbfByteArray
+      case java.lang.Short.TYPE     => cbfShortArray
+      case java.lang.Character.TYPE => cbfCharArray
+      case java.lang.Integer.TYPE   => cbfIntArray
+      case java.lang.Long.TYPE      => cbfLongArray
+      case java.lang.Float.TYPE     => cbfFloatArray
+      case java.lang.Double.TYPE    => cbfDoubleArray
+      case java.lang.Boolean.TYPE   => cbfBooleanArray
+      case java.lang.Void.TYPE      => cbfUnitArray
+      case objectClass              => cbfObjectArray
+      case _ => makeCBF[T with AnyRef]
```

Comment:
Ditto.

Reply:
seemed implied in IDE. Removed and updated objectClass to ObjectClass